File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/ResponseMetadata.java
package work.socialhub.kslack.entity;

import com.google.gson.annotations.SerializedName;

/**
 * An attribute included within the response of the newest Slack APIs which contains a {@code next_cursor}
 * value.  By using this value as a cursor parameter in a subsequent request, along with limit, you may
 * navigate through the collection page by virtual page.
 */
public class ResponseMetadata {

    @SerializedName("next_cursor")
    private String nextCursor;

    public String getNextCursor() {
        return this.nextCursor;
    }

    public void setNextCursor(String nextCursor) {
        this.nextCursor = nextCursor;
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/ResponseMetadata.kt
package work.socialhub.kslack.entity

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.js.JsExport

/**
 * An attribute included within the response of the newest Slack APIs which contains a `next_cursor`
 * value.  By using this value as a cursor parameter in a subsequent request, along with limit, you may
 * navigate through the collection page by virtual page.
 */
@JsExport
@Serializable
class ResponseMetadata {

    @SerialName("next_cursor")
    var nextCursor: String? = null

    // e.g., [ERROR] must be less than 100 [json-pointer:/limit]
    var messages: Array<String>? = null

    // e.g., method_deprecated
    var warnings: Array<String>? = null
}
